scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala

```diff
@@ -96,7 +92,8 @@ object Term {
     checkFields(enums.head.is[Enumerator.Generator])
   }
   @ast class ForYield(enums: List[Enumerator] @nonEmpty, body: Term) extends Term
-  @ast class New(templ: Template) extends Term
+  @ast class New(init: Init) extends Term
+  @ast class NewAnonymous(templ: Template) extends Term
```

Comment:
How about `NewRefinement`? That would be consistent with the language spec https://www.scala-lang.org/files/archive/spec/2.11/03-types.html#compound-types